### PR TITLE
feat: add bls fil_hash_verify for faster verifications

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

File: rust/src/bls/api.rs [hunk 1]
```diff
@@ -175,6 +175,58 @@ pub unsafe extern "C" fn fil_verify(
...
 
+/// Verify that a signature is the aggregated signature of the hhashed messages
+///
+/// # Arguments
+///
+/// * `signature_ptr`             - pointer to a signature byte array (SIGNATURE_BYTES long)
+/// * `messages_ptr`              - pointer to an array containing the pointers to the messages
+/// * `messages_sizes_ptr`        - pointer to an array containing the lengths of the messages
+/// * `messages_len`              - length of the two messages arrays
+/// * `flattened_public_keys_ptr` - pointer to a byte array containing public keys
```
Comment (laser) [inline rust/src/bls/api.rs:186]: @dignifiedquire - nitpick: you're missing a comment for `flattened_public_keys_len` parameter

**Comment by dignifiedquire** [comment]
@laser thoughts on that failure on mac?

**Comment by laser** [comment]
> @laser thoughts on that failure on mac?

bindings were out of date. i just updated your branch

**Comment by dignifiedquire** [comment]
@laser could you resolve those conflicts please, I still can't generated things myself.

**Pull request merged**